### PR TITLE
Fix spherical harmonics parsing

### PR DESCRIPTION
**Pull request by bejado**

### CODE REVIEW

File: libs/image/src/KtxBundle.cpp [hunk 1]
```diff
@@ -292,7 +292,8 @@ bool KtxBundle::getSphericalHarmonics(filament::math::float3* result) {
         return false;
     }
     float* flat = &result->x;
-    for (int i = 0; i < 9; i++) {
+    // 9 bands of 3 coefficients
+    for (int i = 0; i < 9 * 3; i++) {
```
Comment (romainguy) [inline libs/image/src/KtxBundle.cpp]: Could we store them as floats directly in the ktx bundle?

Comment (prideout) [reply libs/image/src/KtxBundle.cpp]: Maybe, but KtxBundle is generic and can read any KTX file.  The sh metadata is a Filament-specific thing, this function was provided as a convenience only.

Comment (romainguy) [reply libs/image/src/KtxBundle.cpp]: I was suggesting this because it's Filament specific ^^

File: libs/image/include/image/KtxBundle.h [hunk 2]
```diff
@@ -106,7 +106,7 @@ class KtxBundle {
     /**
      * Parses the key="sh" metadata and returns 3 bands of data.
      *
-     * Assumes 3 bands for a total of 9 coefficients.
+     * Assumes 9 bands for a total of 27 coefficients.
```
Comment (pixelflinger) [inline libs/image/include/image/KtxBundle.h]: The comment is wrong. 
It was correct before except that 1 coefficient is 3 float values (RGB).
So it's *3* bands, and 27 floats (3^2  RGB coefficients).

File: libs/image/src/KtxBundle.cpp [hunk 3]
```diff
@@ -292,7 +292,8 @@ bool KtxBundle::getSphericalHarmonics(filament::math::float3* result) {
         return false;
     }
     float* flat = &result->x;
-    for (int i = 0; i < 9; i++) {
+    // 9 bands of 3 coefficients
```
Comment (pixelflinger) [inline libs/image/src/KtxBundle.cpp]: 3 bands, 9 RGB coefficients

**Pull request merged**